src/utils/validateUtils.ts

```diff
@@ -40,4 +41,30 @@ export namespace validateUtils {
             return localize('invalidBetweenInputLength', 'The input value must be between {0} and {1} characters long.', lowerLimitIncl, upperLimitIncl);
         }
     }
+
+    /**
+     * Validates that the given input string consists of lower case alphanumeric characters,
+     * starts and ends with an alphanumeric character, and does not contain any special symbols not explicitly specified
+     *
+     * @param value The original input string to validate
+     * @param symbols Any custom symbols that are also allowed in the input string. Defaults to '-'.
+     *
+     * @example
+     * "abcd-1234" // returns true
+     * "-abcd-1234" // returns false
+     */
+    export function isLowerCaseAlphanumericWithSymbols(value: string, symbols: string = '-'): boolean {
```

Comment:
Was thinking we could expand on this with additional options in the future as well.  For example: (1) option to allow uppercase values (would have to rename the function), (2) option to restrict repeating symbols, etc.

Could also add these to shared utils if others find this useful in some capacity.  Open to any thoughts/recommendations :)